src/elements.cpp

```diff
@@ -124,12 +124,8 @@ void G1Element::CheckValid() const {
     ep_set_infty(unity);
     if (g1_cmp(point, unity) != RLC_EQ)
         throw std::invalid_argument("Given G1 element failed in_subgroup check");
-    try {
-        BLS::CheckRelicErrorsInvalidArgument();
-    } catch (...) {
-        throw std::invalid_argument("Relic reports invalid argument given");
-    }
     bn_free(order);
+    BLS::CheckRelicErrors();
```

Comment:
@xdustinface What is the reason that you called this underneath `bn_free` instead of above it like the previous try/catch block?

Reply:
The idea was to ensure `order` gets even freed i.e.  `bn_free` gets called even if `CheckRelicErrors` throws. But seems like it should be moved up 2 more lines to absolutely ensure that.. 🤔